### PR TITLE
add `diag`

### PR DESCRIPTION
**Pull request by putianyi889**

### CODE REVIEW

File: src/ToeplitzMatrices.jl [hunk 1]
```diff
@@ -29,6 +30,7 @@ convert(::Type{AbstractToeplitz{T}}, A::AbstractToeplitz) where T = AbstractToep
 
 isconcrete(A::AbstractToeplitz) = isconcretetype(typeof(A.vc)) && isconcretetype(typeof(A.vr))
 iszero(A::AbstractToeplitz) = iszero(A.vc) && iszero(A.vr)
+diag(A::AbstractToeplitz) = Fill(A.vc[1], min(size(A)...))
```
Comment (jishnub) [inline src/ToeplitzMatrices.jl:33]: This can be done as a `fill`, right? That way, the dependency on `FillArrays` isn't needed (although it's not a bad idea in itself)

Comment (putianyi889) [reply src/ToeplitzMatrices.jl:33]: As of Julia 1.9, how about using `fill` as the minimal feature and override it with `Fill` in the extension?

Comment (jishnub) [reply src/ToeplitzMatrices.jl:33]: We wouldn't want to override a method, that'll hurt precompilation. Extensions are for adding extra methods.

That said, perhaps using `FillArrays` isn't a bad idea after all

Comment (putianyi889) [reply src/ToeplitzMatrices.jl:33]: If both `vc` and `vr` are lazy and large then using `fill` could cause a problem. 

`range(x, x, length=n)` could mimic `Fill(x, n)` though.

**Comment by putianyi889** [comment]
drafted to implement `diag(A,k)`

**Comment by putianyi889** [comment]
also need to test for large `k`. should return empty vector.

**Comment by putianyi889** [comment]
need to test for rectangular ones and negative `k` for Hankel

**Comment by jishnub** [comment]
Looks good to me

**Review by jishnub — APPROVED** [review]
